### PR TITLE
Add SMT threading on Chrysalis

### PR DESCRIPTION
**Pull request by amametjanov**
Add SMT threading on Chrysalis. SMT (hyper-threading) run can be configured by
* `MAX_MPITASKS_PER_NODE=128` in a pureMPI pelayout, or
* `MAX_TASKS_PER_NODE=128` in a multi-threaded pelayout.

Existing pelayouts continue in non-SMT mode.
This PR is just to enable experimenting on SMT pelayouts.

[BFB]

### CODE REVIEW

**Comment by amametjanov** [comment]
SMT is enabled on 20 nodes in `debug` queue.

File: cime_config/machines/config_machines.xml [hunk 1]
```diff
@@ -1145,7 +1145,6 @@
       <executable>srun</executable>
       <arguments>
         <arg name="num_tasks">--mpi=pmi2 -l -n {{ total_tasks }} -N {{ num_nodes }} --kill-on-bad-exit </arg>
-        <arg name="binding">--cpu_bind=cores</arg>
```
Comment (rljacob) [inline cime_config/machines/config_machines.xml:1148]: why is this removed?

Comment (amametjanov) [reply cime_config/machines/config_machines.xml:1148]: It is replaced by identical SLURM_CPU_BIND env-var, just to keep SMT available for MPI-only runs too.

**Comment by amametjanov** [comment]
This PR is required to get SMT ("hyperthreading") working properly on Chrysalis. It was waiting for all nodes to get rebooted  with SMT enabled: this was done during maintenance on Mon, Apr 12. Merging to `next`.
